tests/platforms/linux/appimage/test_build.py

```diff
@@ -390,17 +442,43 @@ def test_build_appimage_with_plugins_in_docker(build_command, first_app, tmp_pat
         os.fsdecode(local_file_plugin_path),
     ]
 
-    # Mock Docker responses
     build_command._subprocess.check_output.return_value = "/docker/bin:/docker/sbin"
 
     # Enable docker, and move to a non-Linux OS.
-    build_command.host_os = "TestOS"
+    build_command.tools.host_os = "TestOS"
     build_command.use_docker = True
 
-    build_command.build_app(first_app)
+    build_command.verify_app_tools(first_app)
 
-    # Ensure that the effect of the Docker context has been reversed.
-    assert build_command.subprocess != build_command.Docker
+    # Docker image is built
+    build_command._subprocess.Popen.assert_any_call(
+        [
+            "docker",
+            "build",
+            "--progress",
+            "plain",
+            "--tag",
+            f"briefcase/com.example.first-app:py3.{sys.version_info.minor}",
+            "--file",
+            f"{tmp_path / 'base_path' / 'linux' / 'appimage' / 'First App' / 'Dockerfile'}",
+            "--build-arg",
+            f"PY_VERSION=3.{sys.version_info.minor}",
+            "--build-arg",
+            "SYSTEM_REQUIRES=",
+            "--build-arg",
+            "HOST_UID=1000",
+            "--build-arg",
+            "HOST_GID=1001",
+            f"{tmp_path / 'base_path' / 'src'}",
+        ],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        bufsize=1,
+        text=True,
+        encoding=mock.ANY,
+    )
```

Comment:
This part of the test becomes redundant once there's an independent test of `verify_app_tools`

Reply:
I removed all the tool verification setup and just provided a `DockerAppContext` directly.